content/guides/development.md

```diff
@@ -5,182 +5,247 @@ contributors:
+  }
 ```
 
-After installing, you can use the middleware like this:
+Now we can run `npm run server` from the command line and we will see our browser automatically loading up a new page (usually `localhost:8080`). If you now change any of the source files and save them, the web server will automatically reload after the code has been compiled. Give it a try!
 
-```js
-var express = require("express");
-var webpackDevMiddleware = require("webpack-dev-middleware");
-var webpack = require("webpack");
-var webpackConfig = require("./webpack.config");
+The `webpack-dev-server` comes with many configurable options. Head over to the [`documentation`](/configuration/dev-server) to learn more.
 
-var app = express();
-var compiler = webpack(webpackConfig);
+T> Now that your server is working, you might want to give [`Hot Module Replacement`](/guides/hot-module-replacement) a try. It's a logical next step!
 
-app.use(webpackDevMiddleware(compiler, {
-  publicPath: "/" // Same as `output.publicPath` in most cases.
-}));
 
-app.listen(3000, function () {
-  console.log("Listening on port 3000!");
-});
-```
+### Using webpack-dev-middleware
 
-Depending on what you've used in `output.publicPath` and `output.filename`, your bundle should now be available on `http://localhost:3000/bundle.js`.
+W> This section is currently not working out of the box.
 
-By default, **watch mode** is used. It's also possible to use **lazy mode**, which will only recompile on a request to the entry point.
 
-To compile only on a request to the entry `bundle.js`:
+## Adjusting Your Text Editor
 
-```js
-app.use(webpackDevMiddleware(compiler, {
-  lazy: true,
-  filename: "bundle.js" // Same as `output.filename` in most cases.
-}));
-```
+When using automatic compilation of your code, you could run into issues when saving your files. Some editors have a "safe write "feature that can potentially interfere with recompilation.
+
+To disable this feature in some common editors, see the list below:
 
-There are many more options you can use. For all configuration options, see the [**devServer documentation**](/configuration/dev-server).
+ * **Sublime Text 3** - Add `atomic_save: "false"` to your user preferences.
+ * **IntelliJ** - use search in the preferences to find "safe write" and disable it.
+ * **Vim** - add `:set backupcopy=yes` to your settings.
+ * **WebStorm** - uncheck Use `"safe write"` in `Preferences > Appearance & Behavior > System Settings`.
 
 
-## References
+ ## Conclusion
 
-* [SurviveJS - Automatic Browser Refresh](http://survivejs.com/webpack/developing-with-webpack/automatic-browser-refresh/)
-* [webpack your Chrome DevTools Workspaces](https://medium.com/@rafaelideleon/webpack-your-chrome-devtools-workspaces-cb9cca8d50da)
+ Now that you've learned how to automatically compile your code whenever you save a file, and how to run a simple development server, you can check out the next guide that covers [`Hot Module Replacement`](/guides/hot-module-replacement).
```

Comment:
You can leave out the second "how to" in this sentence and probably the "whenever you save a file bit" as well...

> Now that you've learned how to automatically compile your code and run a simple development server, you can check out...